packages/svelte/src/compiler/phases/3-transform/client/visitors/RegularElement.js

```diff
@@ -615,11 +628,40 @@ function build_element_attribute_update_assignment(element, node_id, attribute,
 		}
 		return true;
 	} else {
-		state.init.push(update);
+		if (has_expression_tag && can_inline) {
+			push_template_quasi(context.state, ` ${name}="`);
+			push_template_expression(context.state, value);
+			push_template_quasi(context.state, '"');
+		} else {
+			state.init.push(update);
+		}
 		return false;
 	}
 }
 
+/**
+ * @param {(AST.Text | AST.ExpressionTag)[]} nodes
+ * @param {import('../types.js').ComponentClientTransformState} state
+ */
+function can_inline_all_nodes(nodes, state) {
+	let can_inline = true;
+	let has_expression_tag = false;
+	for (let value of nodes) {
+		if (value.type === 'ExpressionTag') {
+			if (value.expression.type === 'Identifier') {
+				const binding = state.scope
+					.owner(value.expression.name)
+					?.declarations.get(value.expression.name);
+				can_inline &&= can_inline_variable(binding);
+			} else {
+				can_inline = false;
+			}
+			has_expression_tag = true;
+		}
+	}
+	return { can_inline, has_expression_tag };
```

Comment:
This returns two values, but at the usage site they're used together - can we simplify this? Also, does it make sense to put this function into the `utils.js` file?

Reply:
I simplified it

I think it probably makes sense to leave it where it is as it isn't used anywhere else. `can_inline_variable` was used multiple places in the predecessor PR where we were hoisting, which is why I put it in the `utils.js` as I expect it will be used elsewhere in the future